pf4j/src/main/java/org/pf4j/AbstractPluginManager.java

```diff
@@ -274,8 +274,7 @@ protected boolean unloadPlugin(String pluginId, boolean unloadDependents) {
         try {
             if (unloadDependents) {
                 List<String> dependents = dependencyResolver.getDependents(pluginId);
-                while (!dependents.isEmpty()) {
-                    String dependent = dependents.remove(0);
+                for (String dependent : dependents) {
                     unloadPlugin(dependent, false);
                     dependents.addAll(0, dependencyResolver.getDependents(dependent));
```

Comment:
I think that this line will produce in the end a `java.util.ConcurrentModificationException`.